#################make all kind of tables for integrative analysis###############

##################################table prep####################################
#delete first column of transcript_level_results_table and gene_level_results_table (contain row number)
#add column name 'target_id' to column which contains ENST... (=transcript name column)
#and column name 'ens_gene' to column which contains ENSG... (=gene name column)

#set working directory
setwd("C:/Users/livbr/bioinformatics/rna_seq")

####################percentage of polyAsite and TSS transcripts#################
##get transcripts which are the same in polyAsite_transcripts and TSS_transcripts
polyAsite_transcripts_file <-read.table(file='polyAsite_transcripts.txt',sep='\t',header=FALSE)
polyAsite_cut <- polyAsite_transcripts_file[,10]
write.table(polyAsite_cut, file='polyAsite_cut.txt', sep=';')
#you might have to delimit with Excel...
polyAsite_cut<-read.table(file='polyAsite_cut.txt', sep='\t')
polyAsite_cut<- polyAsite_cut[,5]

TSS_transcripts_file <- read.table(file='TSS_transcripts.txt',sep='\t', header=FALSE)
TSS_cut<- TSS_transcripts_file[,10]
write.table(TSS_cut, file='TSS_cut.txt',sep=';')
#you might have to delimit with Excel
TSS_cut <- read.table(file='TSS_cut.txt',sep='\t')
TSS_cut<- TSS_cut[,5]

#get transcript ids which are identical in polyAsite and TSS and loosing duplicates
TSS_polyA_cut<-intersect(polyAsite_cut, TSS_cut)
length_TSS_polyA <- length(TSS_polyA_cut)
write.table(TSS_polyA_cut, file='TSS_polyA_cut.txt', sep='\t')
##count transcripts of merged.bed file = transcript assembly and delete duplicates
#prep merged.bed with Excel (one argument per column)
total<- read.table(file='op_merged_all.bed.txt', sep='\t')
total_cut<-total[,13]
total_cut<-total_cut[!duplicated(total_cut)]
lenght_total<-length(total_cut)

##get percentage of transcripts which have correct annotation
percentage_of_corr_transcripts <-length_TSS_polyA/lenght_total*100

###########################estimate coding potential############################
#load dataframes (generated by differential_expression_sleuth.R and integrative_analysis_CPAT.R)
transcript_result_table_file <- read.table(file="Transcript_level_result_table.txt", sep="\t", header=TRUE)
gene_result_table_file <- read.table(file="Gene_level_result_table.txt", sep='\t',header=TRUE)
coding_potential_file <- read.table(file="coding_potential.txt",sep="\t", header=TRUE)

#merge dataframes
results_table<-merge(x=transcript_result_table_file,y=gene_result_table_file,by='ens_gene',all.x=TRUE, all.y=TRUE)
merged_results_table<-merge(x=results_table,y=coding_potential_file,by='target_id', all.x=TRUE, all.y=TRUE)
write.table(merged_results_table, file='merged_results_table.txt', sep='\t')

##in Excel:
#replace 'NA' with '#N/A' to make it readable for Excel
#to clean the potential from #N/A values use: =IFNA($cell,0) in a new column
#the coding potential threshold is 0.364
#we want to use 0 and 1 to give information about the coding potential
#to do so use: if($cell<0.364,0,1) in a new column (take the cells from the cleaned potential)
#to get the percentage of coding novel transcripts out of all novel transcripts, filter
#for 'MSTRG.NUMBER'. Divide number of novel coding transcripts by total number of novel transcripts *100


###############################get final table##################################
#merge merged_results_table.txt, TSS_polyA_cut.txt and intergenic_transcripts.txt 
#into one table

#use Excel to filter for novel transcipts (=MSTRG.NUMBER) in the merged_results_table.txt
#and make new table out of it (handier, as it is way smaller)

#delete first column of TSS_polyA_cut.txt (contains row number) and name the remaining column as 'target_id'
#filter for novel transcripts and add a new column called 'correct_end_annotation' and fill the whole columne with '1'
#copy the current content to a new excel file

#use Excel to delimite intergenic_transcripts.txt
#delete all columns except for the one containing the transcript ids (=column M)
#delete all duplicates and name column 'target_id'
#add '1' next to the column 'target_id' (so we can directly see which transcripts
#are intergenic in the final table, name this column 'intergenic'
#save the file as mod_intergenic_transcripts.xlsx

#load tables and library to read excel files
library("readxl")
novel_transcripts_file <- read_excel('novel_transcripts.xlsx')
TSS_polA_cut_file<- read_excel('TSS_polyA_novel_only.xlsx')
intergenic_cut_file <- read_excel('mod_intergenic_transcripts.xlsx')

#merge tables
novel_trans_polyATSS<-merge(x=novel_transcripts_file,y=TSS_polA_cut_file,by='target_id', all.x=TRUE)
novel_polyATSS_intergenic<-merge(x=novel_trans_polyATSS, y=intergenic_cut_file, by='target_id', all.x=TRUE)
write.table(novel_polyATSS_intergenic, file='novel_transcripts_all_info.txt', sep='\t')

